diff --git a/media/module/foundation/AHandler.cpp b/media/module/foundation/AHandler.cpp
index d8b0aaf..9cea9fb 100644
--- a/media/module/foundation/AHandler.cpp
+++ b/media/module/foundation/AHandler.cpp
@@ -41,14 +41,14 @@ void AHandler::deliverMessage(const sp<AMessage> &msg) {
 }

 void AHandler::setDeliveryStatus(bool delivering, uint32_t what, int64_t startUs) {
-    AutoMutex autoLock(mLock);
+    //AutoMutex autoLock(mLock);
     mDeliveringMessage = delivering;
     mCurrentMessageWhat = what;
     mCurrentMessageStartTimeUs = startUs;
 }

 void AHandler::getDeliveryStatus(bool& delivering, uint32_t& what, int64_t& durationUs) {
-    AutoMutex autoLock(mLock);
+    //AutoMutex autoLock(mLock);
     delivering = mDeliveringMessage;
     what = mCurrentMessageWhat;
     durationUs = mCurrentMessageStartTimeUs == 0 ?
diff --git a/services/audioflinger/Effects.cpp b/services/audioflinger/Effects.cpp
index c73b946..276698f 100644
--- a/services/audioflinger/Effects.cpp
+++ b/services/audioflinger/Effects.cpp
@@ -2900,6 +2900,11 @@ static const effect_uuid_t SL_IID_VOLUME_ = { 0x09e8ede0, 0xddde, 0x11db, 0xb4f6
 const effect_uuid_t * const SL_IID_VOLUME = &SL_IID_VOLUME_;
 #endif //OPENSL_ES_H_

+// Dolby Atmos
+static const effect_uuid_t SL_IID_DAP_ = // 46d279d9-9be7-453d-9d7c-ef937f675587
+{ 0x46d279d9, 0x9be7, 0x453d, 0x9d7c, {0xef, 0x93, 0x7f, 0x67, 0x55, 0x87} };
+const effect_uuid_t * const SL_IID_DAP = &SL_IID_DAP_;
+
 /* static */
 bool EffectChain::isEffectEligibleForBtNrecSuspend_l(const effect_uuid_t* type) {
     // Only NS and AEC are suspended when BtNRec is off
@@ -2917,7 +2922,8 @@ bool EffectChain::isEffectEligibleForSuspend(const effect_descriptor_t& desc)
         (((desc.flags & EFFECT_FLAG_TYPE_MASK) == EFFECT_FLAG_TYPE_AUXILIARY) ||
          (memcmp(&desc.type, SL_IID_VISUALIZATION, sizeof(effect_uuid_t)) == 0) ||
          (memcmp(&desc.type, SL_IID_VOLUME, sizeof(effect_uuid_t)) == 0) ||
-         (memcmp(&desc.type, SL_IID_DYNAMICSPROCESSING, sizeof(effect_uuid_t)) == 0))) {
+         (memcmp(&desc.type, SL_IID_DYNAMICSPROCESSING, sizeof(effect_uuid_t)) == 0) ||
+         (memcmp(&desc.type, SL_IID_DAP, sizeof(effect_uuid_t)) == 0))) {
         return false;
     }
     return true;
